### PR TITLE
Separate map component

### LINKED ISSUE #1551: Add geometry for feature with null geometry
Right now, if Input creates a new linked feature, it is created with null geometry. For spatial tables, it would be very useful to have a button "add geometry" in the attribute form, which would show the map and start digitizing of the feature's geometry.

### PR DESCRIPTION
**Pull request by tomasMizera**
Separated map component with all its relative components (compass, position kit, digitizing controller, toolbar,..) to subfolder called `map`. There is a new qml type called `MapWrapper.qml` that wraps map functionalities like showing map, identifying features and 
recording geometry.

This PR do not change anything visually in app.

Note:
 - This separation was must in order to close #1551 
 - Currently it is not possible to edit geometry, will be included in next PR together with adding geometry for new linked features
 - I had to add all icons to `InputStyle` with `qrc:/` prefix because qml files moved

### CODE REVIEW

File: app/qml/map/LoadingIndicator.qml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import QtQuick 2.7
 import QtQuick.Controls 2.2
 import QtGraphicalEffects 1.0
-
+import ".."
```
Comment (sklencar) [inline app/qml/map/LoadingIndicator.qml:13]: Does it need such import?

File: app/qml/map/MapWrapper.qml [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+  function clear() {
+    // clear all previous references to old project (if we don't clear references to the previous project,
+    // highlights may end up with dangling pointers to map layers and cause crashes)
+    _highlightIdentified.featureLayerPair = null
+    _digitizingHighlight.featureLayerPair = null
+  }
+
+  states: [
+    State {
+      name: "View"
```
Comment (sklencar) [inline app/qml/map/MapWrapper.qml:168]: We used to name states starting with small letter as well as can be seen [here. in coding conventions](https://doc-snapshots.qt.io/qt6-dev/qml-codingconventions.html).
Any particular reason while states starts with a capital letter?

File: app/qml/map/MapWrapper.qml [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+
+      mapSettings: _map.mapSettings
+      identifyMode: IdentifyKit.TopDownAll
+    }
+
+    onIsRenderingChanged: _loadingIndicator.visible = isRendering
+
+    onClicked: {
+      if ( !root.isInRecordState ) {
+        //       mapCanvas.forceActiveFocus()
```
Comment (sklencar) [inline app/qml/map/MapWrapper.qml:251]: probably leftover?

File: app/qml/map/MapWrapper.qml [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+  }
+
+  StateGroup {
+    id: _gpsState
+
+    property color indicatorColor
+
+    states: [
+      State {
+        name: "Good"
```
Comment (sklencar) [inline app/qml/map/MapWrapper.qml:300]: Same as naming for _map states

File: app/qml/map/MapWrapper.qml [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+          isGpsWorking      &&
+          isUsingPosition
+    }
+
+    width: parent.width - _gpsAccuracyBanner.anchors.margins * 2
+    height: InputStyle.rowHeight * 2
+
+    text: qsTr( "Low GPS position accuracy (%1 m)<br><br>Please make sure you have good view of the sky." )
+    .arg( __inputUtils.formatNumber( _positionKit.accuracy ) )
+    link: "https://help.inputapp.io/howto/gps_accuracy"
```
Comment (sklencar) [inline app/qml/map/MapWrapper.qml:446]: Better to add it to InputHelp among other links

File: app/qml/qml.qrc [hunk 6]
```diff
@@ -79,16 +72,24 @@
...
         <file>form/FeatureToolbar.qml</file>
         <file>form/PreviewPanel.qml</file>
         <file>form/FormWrapper.qml</file>
         <file>form/FeatureFormPage.qml</file>
-        <file>editor/RelationTextDelegate.qml</file>
-        <file>editor/RelationPhotoDelegate.qml</file>
-        <file>editor/RelationPhotoFooterDelegate.qml</file>
-        <file>editor/inputrelationreference.qml</file>
-        <file>editor/AbstractEditor.qml</file>
+        <file>map/MapWrapper.qml</file>
```
Comment (sklencar) [inline app/qml/qml.qrc:86]: Since introducing new subfolder (and generally redefining the list), could be good to have it sorted alphabetically.

**Review by sklencar — CHANGES_REQUESTED** [review]
Good job, generally looks good, just small notes.

However I found a bug related to digitizing - while changing map themes and visible layers, it could happen that no digitizing layer is selected and therefore Input crash while adding a point.

To replicate it with input-testing/tc04_recording:
1. Open the project and click on record button
2. Make sure that (point) layer is selected as digitizng layer and a point can be created
3. Change a map theme from `Points` to `Paths and polygons`. 
4. Click on the record button and check if any layer for digitizing is selected -> here I can see no layer is selected and therefore the app crashed after adding a point.

**Comment by sklencar** [comment]
Additionally, icons are broken:
Account page is missing icons with error message(s): 
`qrc:/components/TextWithIcon.qml:35:5: QML Image: Cannot open: qrc:/components/account.svg`

Screenshot 2021-08-03 at 15 38 09

File: app/qml/map/LoadingIndicator.qml (hunk 1 above)
Comment (tomasMizera) [reply app/qml/map/LoadingIndicator.qml:13]: Yes, for InputStyle

File: app/qml/InputStyle.qml [hunk 7]
```diff
@@ -65,41 +65,56 @@ QtObject {
...
-    property var detachIcon: "qrc:/unlink.svg"
-    property var valueRelationIcon: "qrc:/value_relation_open.svg"
-    property var comboboxIcon: "qrc:/combobox.svg"
-    property var qrCodeIcon: "qrc:/qrcode.svg"
-    property var exclamationIcon: "qrc:/exclamation-circle.svg"
-    property var syncIcon: "qrc:sync.svg"
-    property var downloadIcon: "qrc:download.svg"
-    property var stopIcon: "qrc:stop.svg"
-    property var moreMenuIcon: "qrc:/more_menu.svg"
+    property string cameraIcon: "qrc:/add_photo.svg"
```
[the remark this answers is not in the document]
Comment (tomasMizera) [reply app/qml/InputStyle.qml:72]: Tried to divide it into related categories

**Comment by tomasMizera** [comment]
@sklencar all notes are resolved

**Review by sklencar — APPROVED** [review]

**Pull request merged**